core/models/project.py

```diff
@@ -89,7 +89,7 @@ def add_object(self, related_obj):
         new_join.save()
         return new_join
 
-    def _test_project_ownership(self, user):
+    def _test_project_ownership(self, user, related_obj):
```

Comment:
on line 98 it references related_obj.  This function is only called from the three places above where there is a different related object, so added related_obj to the call signature.